viewing_party/party.py

```diff
@@ -1,23 +1,180 @@
 # ------------- WAVE 1 --------------------
 
+from tests.test_constants import MOVIE_TITLE_1
+
+movie_dictionary = {}
+
 def create_movie(title, genre, rating):
-    pass
+    if title and genre and rating:
+        movie_dictionary["title"] = title
+        movie_dictionary["genre"] = genre
+        movie_dictionary["rating"] = rating
+        return movie_dictionary
+    else:
+        return None
+
+def add_to_watched(user_data, movie):
+    user_data = {"watched":[movie]}
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data = {"watchlist": [movie]}
+    return user_data
+
+def watch_movie(user_data, title):
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)
+            return user_data
+    else:
+            return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    total_rating = 0
+    index = 0 
+    if user_data == {"watched": []}:

```

Comment:
This check passes our tests, but it will fail in cases where there is more data in the `user_data` object.  Checking `user_data["watched"]` to see if it is falsy or empty avoids this bug.
```suggestion
    if not user_data["watched"]:

```